### PR TITLE
Fix jca error handling

### PR DESCRIPTION
**Pull request by n0900**
Consistent error handling as described in the README + small clean up

### CODE REVIEW

File: datatypes/src/jvmMain/kotlin/at/asitplus/crypto/datatypes/JcaExtensions.kt [hunk 1]
```diff
@@ -75,29 +75,31 @@ fun CryptoPublicKey.Rsa.getPublicKey(): RSAPublicKey =
...
 
+@Throws(Throwable::class)
 fun CryptoPublicKey.Companion.fromJcaKey(publicKey: PublicKey): KmmResult<CryptoPublicKey> =
     runCatching {
         when (publicKey) {
-            is RSAPublicKey -> CryptoPublicKey.Rsa.fromJcaKey(publicKey)
-            is ECPublicKey -> CryptoPublicKey.Ec.fromJcaKey(publicKey)
+            is RSAPublicKey -> CryptoPublicKey.Rsa.fromJcaKey(publicKey).getOrThrow()
+            is ECPublicKey -> CryptoPublicKey.Ec.fromJcaKey(publicKey).getOrThrow()
             else -> throw IllegalArgumentException("Unsupported Key Type")
```
Comment (JesusMcCloud) [inline datatypes/src/jvmMain/kotlin/at/asitplus/crypto/datatypes/JcaExtensions.kt:103]: don't `getOrThrow()` just to wrap it: better just return it (as is is a KmmResult already) and explicitly return a failure for the `else` branch

File: datatypes/src/jvmTest/kotlin/Pkcs10CertificationRequestJvmTest.kt [hunk 2]
```diff
@@ -37,7 +37,7 @@ class Pkcs10CertificationRequestJvmTest : FreeSpec({
 
     "CSR match" {
         val ecPublicKey = keyPair.public as ECPublicKey
-        val cryptoPublicKey = CryptoPublicKey.Ec.fromJcaKey(ecPublicKey)
+        val cryptoPublicKey = CryptoPublicKey.Ec.fromJcaKey(ecPublicKey).getOrThrow()
         cryptoPublicKey.shouldNotBeNull()
```
Comment (JesusMcCloud) [inline datatypes/src/jvmTest/kotlin/Pkcs10CertificationRequestJvmTest.kt:41]: requires a `getOrNull()` in line 40 for `shouldNotBeNull()` to make sense

File: datatypes/src/jvmTest/kotlin/Pkcs10CertificationRequestJvmTest.kt [hunk 3]
```diff
@@ -77,7 +77,7 @@ class Pkcs10CertificationRequestJvmTest : FreeSpec({
 
     "CSR with attributes match" {
         val ecPublicKey = keyPair.public as ECPublicKey
-        val cryptoPublicKey = CryptoPublicKey.Ec.fromJcaKey(ecPublicKey)
+        val cryptoPublicKey = CryptoPublicKey.Ec.fromJcaKey(ecPublicKey).getOrThrow()
         cryptoPublicKey.shouldNotBeNull()
```
Comment (JesusMcCloud) [inline datatypes/src/jvmTest/kotlin/Pkcs10CertificationRequestJvmTest.kt:81]: requires a `getOrNull()`  in line 80 for `shouldNotBeNull()` to make sense

File: datatypes/src/jvmTest/kotlin/Pkcs10CertificationRequestJvmTest.kt [hunk 4]
```diff
@@ -123,7 +123,7 @@ class Pkcs10CertificationRequestJvmTest : FreeSpec({
 
     "CSRs with extensionRequest match" {
         val ecPublicKey = keyPair.public as ECPublicKey
-        val cryptoPublicKey = CryptoPublicKey.Ec.fromJcaKey(ecPublicKey)
+        val cryptoPublicKey = CryptoPublicKey.Ec.fromJcaKey(ecPublicKey).getOrThrow()
         cryptoPublicKey.shouldNotBeNull()
```
Comment (JesusMcCloud) [inline datatypes/src/jvmTest/kotlin/Pkcs10CertificationRequestJvmTest.kt:127]: same

File: datatypes/src/jvmTest/kotlin/PublicKeyTest.kt [hunk 5]
```diff
@@ -38,7 +38,7 @@ class PublicKeyTest : FreeSpec({
                 keys
             ) { pubKey ->
 
-                val own = CryptoPublicKey.Ec.fromJcaKey(pubKey)
+                val own = CryptoPublicKey.Ec.fromJcaKey(pubKey).getOrThrow()
                 own.shouldNotBeNull()
```
Comment (JesusMcCloud) [inline datatypes/src/jvmTest/kotlin/PublicKeyTest.kt:42]: same

File: datatypes/src/jvmTest/kotlin/X509CertificateJvmTest.kt [hunk 6]
```diff
@@ -44,7 +44,7 @@ class X509CertificateJvmTest : FreeSpec({
 
     "Certificates match" {
         val ecPublicKey = keyPair.public as ECPublicKey
-        val cryptoPublicKey = CryptoPublicKey.Ec.fromJcaKey(ecPublicKey)
+        val cryptoPublicKey = CryptoPublicKey.Ec.fromJcaKey(ecPublicKey).getOrThrow()
         cryptoPublicKey.shouldNotBeNull()
```
Comment (JesusMcCloud) [inline datatypes/src/jvmTest/kotlin/X509CertificateJvmTest.kt:48]: same

**Review by JesusMcCloud — CHANGES_REQUESTED** [review]

File: datatypes/src/jvmTest/kotlin/Pkcs10CertificationRequestJvmTest.kt (hunk 2 above)
Comment (n0900) [reply datatypes/src/jvmTest/kotlin/Pkcs10CertificationRequestJvmTest.kt:41]: True, will remove the shouldNotBeNull()

**Review by JesusMcCloud — APPROVED** [review]

**Pull request merged**